### PR TITLE
implement OpenSceneGraph reader from Jason Beverage

### PR DESCRIPTION
**Pull request by hobu**

### CODE REVIEW

File: plugins/openscenegraph/io/OSGReader.cpp [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+static PluginInfo const s_info = PluginInfo(
+"readers.osg",
+"osg reader.",
+"" );
+
+CREATE_SHARED_PLUGIN(1, 0, OSGReader, Reader, s_info)
+
+std::string OSGReader::getName() const { return s_info.name; }
+
+void OSGReader::addArgs(ProgramArgs& args)
```
Comment (abellgithub) [inline plugins/openscenegraph/io/OSGReader.cpp]: This function can go away.

File: plugins/openscenegraph/io/OSGReader.cpp [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+void OSGReader::addDimensions(PointLayoutPtr layout)
+{
+    layout->registerDim(Dimension::Id::X);
+    layout->registerDim(Dimension::Id::Y);
+    layout->registerDim(Dimension::Id::Z);
+}
+
+Dimension::IdList OSGReader::getDefaultDimensions()
```
Comment (abellgithub) [inline plugins/openscenegraph/io/OSGReader.cpp]: This function can go away unless you need it for some reason other than PDAL.

File: plugins/openscenegraph/io/OSGReader.cpp [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    osg::ref_ptr< osg::Node > node = osgDB::readNodeFile(m_filename);
+
+    CollectTrianglesVisitor v;
+    node->accept(v);
+    _points = v._points;
+}
+
+point_count_t OSGReader::read(PointViewPtr view, point_count_t count)
+{
+    PointLayoutPtr layout = view->layout();
```
Comment (abellgithub) [inline plugins/openscenegraph/io/OSGReader.cpp]: Unused.

File: plugins/openscenegraph/io/OSGReader.cpp [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    PointId nextId = view->size();
+    PointId idx = m_index;
+    point_count_t numRead = 0;
```
Comment (abellgithub) [inline plugins/openscenegraph/io/OSGReader.cpp]: Don't think numRead is really ever used.

File: plugins/openscenegraph/io/OSGReader.hpp [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        Reader(),
+        m_index(0)
+    {
+    };
+
+    static void * create();
+    static int32_t destroy(void *);
+    std::string getName() const;
+
+    static Dimension::IdList getDefaultDimensions();
```
Comment (abellgithub) [inline plugins/openscenegraph/io/OSGReader.hpp]: This can go away.

File: plugins/openscenegraph/io/OSGReader.hpp [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+  private:
+    point_count_t m_index;
+
+    std::vector< osg::Vec3d > _points;
+
+    virtual void addDimensions(PointLayoutPtr layout);
+    virtual void addArgs(ProgramArgs& args);
```
Comment (abellgithub) [inline plugins/openscenegraph/io/OSGReader.hpp]: This can go away.

File: plugins/openscenegraph/io/OSGReader.cpp (hunk 2 above)
Comment (hobu) [inline plugins/openscenegraph/io/OSGReader.cpp]: This must be coming from an example somewhere.

Comment (abellgithub) [inline plugins/openscenegraph/io/OSGReader.cpp]: Yes.  I'll search to see what I can find.

**Pull request merged**